refactor/refactor_test.go

```diff
@@ -0,0 +1,337 @@
+package refactor
```

Comment:
Table-driven tests would be nice here. I think it would make the cases much more readable. We don't have to change them, just something to consider in future.